lexer/impl/lexer.cpp

```diff
@@ -1,6 +1,309 @@
+                case '*':
+                case '(':
+                case ')':
+                case '[':
+                case ']':
+                case '=':
+                case ';':
+                case ',':
+                    sign = toStr(c);
+                    val = trie.Find(sign).value_or(invalidToken);
+                    result.push_back(makeToken(val, lineNum, sign));
+                    break;
+                case '.':
+                    checkTwoCharToken(&index, word, "..", TokenType::Dot, TokenType::TwoDots);
+                    break;
+                case '<':
+                    checkTwoCharToken(&index, word, "<=", TokenType::SmallerComp, TokenType::SeqComp);
+                    break;
+                case '>':
+                    checkTwoCharToken(&index, word, ">=", TokenType::BiggerComp, TokenType::BeqComp);
+                    break;
+                case '/':
+                    checkTwoCharToken(&index, word, "/=", TokenType::DivOp, TokenType::NeqComp);
+                    break;
+                case ':':
+                    checkTwoCharToken(&index, word, ":=", TokenType::Column, TokenType::Assignment);
+                    break;
+
+                default:
+
+                    if (isdigit(c))
+                    { // if number constant
+                        index--;
+                        std::string num = readWhileNum(&index, word);
+                        TokenType type = TokenType::IntegerLiteral;
+                        if (index < int(word.length()) && word[index] == '.')
+                        { // if real
+                            int oldIndex = index;
+                            index++;
+                            std::string floatPart = readWhileNum(&index, word);
+                            if (floatPart.length() < 1)
+                            { //eg ..
+                                index = oldIndex;
+                                result.push_back(makeToken(type, lineNum, num));
+                                break;
+                            }
+                            num.append(".");
+                            num.append(floatPart);
+                            type = TokenType::RealLiteral;
+                        }
+                        result.push_back(makeToken(type, lineNum, num));
+                    }
+                    else if (iswalpha(c))
+                    { // if var name
+                        std::string name = toStr(c);
+                        while (index < int(word.length()) && (iswalpha(word[index]) || isdigit(word[index]) || word[index] == '_'))
+                        {
+                            name.append(toStr(word[index++]));
+                        }
+                        result.push_back(makeToken(TokenType::Identifier, lineNum, name));
+                    }
+                    else
+                    { // non-valid construction
+                        std::cout << "INVALID CHAR" << c << "\n";
```

Comment:
DO NOT `cout` in a library code! Propagate errors through the library, handle errors in the user code.